### PR TITLE
Add github CI integration

### PR DESCRIPTION
**Pull request by psibi**
### Description

Replaces Travis CI with github actions since travis doesn't work anymore: https://news.ycombinator.com/item?id=25338983

### Checklist

  - [ ] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I've confirmed these changes don't belong in xmonad-contrib instead

  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [ ] I updated the `CHANGES.md` file

### CODE REVIEW

**Comment by psibi** [comment]
I will merge this in a couple of days unless there is objections. CC: @byorgey and others.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  build:
+    name: CI
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        resolver: [lts-16, lts-15, lts-14]
```
Comment (liskin) [inline .github/workflows/tests.yml]: We claim support all the way back to GHC 8.0 in xmonad.cabal, so we should probably test with lts-7 or 9 and lts-11 and lts-12 as well.

Comment (psibi) [reply .github/workflows/tests.yml]: Sure, we can probably. I will update the resolvers section tomorrow to see if the CI passes. If it doesn't pass, I wouldn't be motivated to fix them though (via a separate stack.yaml file) and would likely remove them. :grinning:

Comment (liskin) [reply .github/workflows/tests.yml]: Debian Stable has 8.4.4 so I'd probably attempt to support lts-12+ but anything older can probably be dropped?

Oh and EPEL8 has ghc 8.2.2. How likely is it someone's running EL8 on desktop and uses xmonad and needs new stuff to work?

Comment (psibi) [reply .github/workflows/tests.yml]: > Debian Stable has 8.4.4 so I'd probably attempt to support lts-12+ but anything older can probably be dropped?

Yeah, I'm leaning towards the same. Sticking to support last three major version of ghc would likely make things simpler. I just added `lts-12` and the CI passes.

**Pull request merged**